### ISSUE TITLE
Installation stuck at starting kubelet

### ISSUE THREAD

**Issue by enlinxu** [coreos/tectonic-forum#26]
My tectonic AWS installation got stuck at Starting Kubelet stage, although the cloudformation says the installation is successful.

I do see the following log in tectonic installer:

Failed to create alias: ValidationException: 1 validation error detected Value 'alians/enlin.xu@vmturbo.com-tectonic/9864' at 'aliasName' failed to satisfy constraint...

Anyone experienced the same?

**Comment by danieloliveira079** [comment]
Something similar happening here. The output log is:
```
Jan 20 14:50:24 ip-10-0-0-50.eu-central-1.compute.internal kubelet-wrapper[1504]: E0120 14:50:24.396470    1504 reflector.go:203] pkg/kubelet/kubelet.go:403: Failed to list *api.Nod
```
Looks like it is an issue related to the assets. I could not find a solution yet. Doesn't make any sense start the Tectonic installer and have to ssh into a server to generate or copy the assets files.

Using AWS eu-central-1

**Comment by vipinsthename** [comment]
I have been facing the same issue with eu-west-1.

bootkube doesn't appear to start. I have tried exploding the assets and starting it up manually. No luck.

<img width="386" alt="screen shot 2017-01-31 at 14 12 11" src="https://cloud.githubusercontent.com/assets/5746019/22469541/429f4f44-e7c4-11e6-8f32-e3a09ceed2c0.png">

**Comment by vipinsthename** [comment]
I found the following fixes it for me:

scp assets-.zip core@<CONTROLLER_DNS_NAME>:/home/core/ &&
ssh core@<CONTROLLER_DNS_NAME> 'unzip assets-.zip \
&& sudo mv assets /opt/bootkube/assets \
&& sudo chmod +x /opt/bootkube/assets/bootkube-start \
&& sudo systemctl start bootkube'

But the provisioner still remains stuck, but you can now access the tectonic console.

**Comment by pompomJuice** [comment]
This is unacceptable. 

How hard can it be to not have fixed this by now. 

Wasted 2 days trying to find out why my oneshot tectonic awesome installer is completely DOA.

This is highly annoying.

**Comment by InAnimaTe** [comment]
Yeah I'm not sure why these issues aren't being tackled a little better. For a project thats been around since mid-2015, it seems wildly immature. Documentation is extremely lacking, response times to issues like this are abysmal, and implementation/installation is still really awkward, especially for bare-metal setups.

Really hoping to see more effort from the team. At this point, I'm not really seeing much if any benefit of using tectonic over vanilla kubernetes (or something deployed via something like Rancher).

**Comment by fii** [comment]
Seeing the same issue ...

**Comment by mfburnett** [comment]
Hey everyone - @enlinxu, thanks for filing this issue, and so sorry for the delayed response! This completely slipped by me :( Will look into this and get back to you all asap. Thanks!

**Comment by sym3tri** [comment]
@enlinxu sorry for the delay. We haven't yet reproduced your error but we suspect what happened is this:

In the installer you opted to create a new KMS key, and the alias creation part of that operation failed due to a validation problem. This could be a bug with how we generate the alias name. If they key was not created properly, the cluster will never boot. This could also be a bug (installer should not allow you to proceed if the key creation fails).

Instead of creating the key via the installer, can you try manually creating one in the AWS Console and then select it from the installer dropdown?

**Comment by enlinxu** [comment]
@sym3tri Thanks for the replay. Reuse existed KMS fix the exception I see in the log, but I still got stuck at starting kubelet.  Not sure about others in this thread.

**Comment by pompomJuice** [comment]
Basically what you need to do is download the matchbox head repo, use contrib to generate k8 keys and feed those to tectonic. 

Then you copy the assets to your controller and restart bootkube.

It should then get to the point where I am now where that is... it looks like the cluster has booted but i cannot find the tectonic console and I am not sure why there is neither a console or kubectl installed. 

What am I supposed to do?

**Comment by pompomJuice** [comment]
Basically what you need to do is download the matchbox head repo, use
contrib to generate k8 keys and feed those to tectonic.

Then you copy the assets to your controller and restart bootkube.

It should then get to the point where I am now where that is... it looks
like the cluster has booted but i cannot find the tectonic console and I am
not sure why there is neither a console or kubectl installed.

What am I supposed to do?

**Comment by lorecrafting** [comment]
Stuck on Starting Kubelet also.  Can anybody shed some light on this situation? It's not even resolving the DNS.  However in AWS cloud formation, it says all the resources and formation is all deployed.
![screen shot 2017-02-11 at 7 46 22 pm](https://cloud.githubusercontent.com/assets/4595918/22859965/f4ec8db6-f092-11e6-8661-295e2695a692.png)

**Comment by m-mikula** [comment]
Hey there I got the same Problem. 

![screenshot from 2017-02-15 13-29-30](https://cloud.githubusercontent.com/assets/13526351/22979870/259e6396-f398-11e6-84ab-a25d9226ba72.png)

Also i don't get any log messages to help with, what could I do to provide any useful information?

**Comment by torstello** [comment]
I also chose the installer to generate the KMS key and it stuck at 'starting kubelet'

I scp'd the assets.zip and startet the bootkube manually on the controller instance.
Before that /opt/bootkube/assets folder was missing.

still stuck.

What I see on my terminal, from which I started the installer (darwin):
`Starting Tectonic Installer on 127.0.0.1:4444`
`ERRO[0175] &{AuthFailure: AWS was not able to validate the provided access credentials`
`status code: 401, request id: 965315e9-b117-4936-8b62-24c951c89a06 AWS was `
`not able to validate the provided access credentials 401}`
`ERRO[0205] &{<nil> Request body is empty. 400}`
`ERRO[0206] &{<nil> Request body is empty. 400}`

on the controller systemctl status bootkube shows a lot of:
`bootkube-start[2683]:... E0221 13:28:40.804878 5 leaderelection.go:228] \`
`error retrieving resource lock kube-system /kube-scheduler:  \`
`the server cannot complete the requested operation \`
`at this time, try again later (get endpoints kube-scheduler)`

and finally 
`bootkube-start[2683]: [ 3136.366755] bootkube[5]: \`
`error while checking pod status: timed out waiting for the condition`

**Comment by pompomJuice** [comment]
@torstello  AWS or bare metal? I presume AWS?

Looks like you have DNS issues.

**Comment by m-mikula** [comment]
So did i also have dns issues? is there any way to activate logging or something? 
I'd be glad to help, but i honestly don't know how to help and what to do.

**Comment by torstello** [comment]
@pompomJuice  yes AWS.
the tectonic installer machine can resolve the URL.
Ping from all instances I'll check with the next try, I already stopped the ec2 instances from today's attempt. I'll keep you informed

**Comment by mfburnett** [comment]
Hey all (cc @enlinxu)! We recently released Tectonic v1.6.2-tectonic.1 - would recommend you check it out! One of the biggest changes is that the AWS installer is now backed by Terraform instead of CloudFormation. Let us know if you're still seeing this issue with the latest installer.

**Comment by rajholla** [comment]
@mfburnett  Do you know when the windows installer for Tectonic v1.6.2-tectonic.1 will be fixed ? 
#136 
I hit the same problem with the older version and like to try the new version.

**Comment by robszumski** [comment]
@rajholla Check out this guide for using the installer on Windows via docker: https://coreos.com/tectonic/docs/latest/install/installer-windows.html
